pkg/reconciler/clusteringress/clusteringress.go

```diff
@@ -126,7 +127,16 @@ func (c *Reconciler) Init(ctx context.Context, cmw configmap.Watcher, impl *cont
 		DeleteFunc: func(obj interface{}) {
 			vs, ok := obj.(*v1alpha3.VirtualService)
 			if ok {
-				statusProber.Cancel(vs)
+				statusProber.CancelVirtualServiceProbing(vs)
+			}
+		},
+	})
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
Should the event scope be somewhat reduced to only the pods that matter, i.e. the gateway pods?

Reply:
I guess it would be somewhat conceptually cleaner.
That being said, CancelPodProbing is just a map lookup and a no-op if the pod is not being probed, so it as lightweight as it can be.
Also, it is not trivial to filter at that location because to determine if a pod is a Gateway pod we need to use the `spec.selector` fields from all Gateways. It is not lightweight at all.